nomos-da/network/messages/src/packing.rs

```diff
@@ -0,0 +1,156 @@
+// STD
+use futures::AsyncReadExt;
+use std::io;
+// Crates
+use nomos_core::wire;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+// Internal
+use crate::Result;
+
+type LenType = u16;
+const MAX_MSG_LEN_BYTES: usize = size_of::<LenType>();
+const MAX_MSG_LEN: usize = 1 << (MAX_MSG_LEN_BYTES * 8);
+
+fn into_failed_to_serialize(error: wire::Error) -> io::Error {
+    io::Error::new(
+        io::ErrorKind::InvalidData,
+        format!("Failed to serialize message: {}", error),
+    )
+}
+
+fn into_failed_to_deserialize(error: wire::Error) -> io::Error {
+    io::Error::new(
+        io::ErrorKind::InvalidData,
+        format!("Failed to deserialize message: {}", error),
+    )
+}
```

Comment:
Doesn't let me, as `wire::Error` is an alias for `bincode::Error`; both types are foreign. Is there an appropriate workaround?

In the meantime I'll remove the duplicate.

Reply:
Yeah, probably then we would have to make `wire::Error` a wrapper type instead of a type alias. Something to be done in a different PR probably. Could you please open an issue for it as well?